### PR TITLE
RSDK-8767 - populate local_name and remote_path to all resource names sent from RDK

### PR DESCRIPTION
**Pull request by Kschappacher**
### Feature
- We want to include additional fields in a resources name object. These additional fields make it easier to parse what remote the resource is on and the local name of the resource
- This feature should help make interacting with named resources easier

### Testing
- Tested changes on the python sdk and typescript sdk

### CODE REVIEW

File: go.mod
Comment (Kschappacher) [inline go.mod:6]: Temp until api gets merged. DO NOT MERGE WITH THIS IN

**Review by cheukt — APPROVED** [review]
LGTM

**Pull request merged**
